### ISSUE TITLE
"Unable to load Vips"; ...; "Check that the typelib `Vips-8.0.typelib` has been installed, and that it is on your GI_TYPELIB_PATH."?

### ISSUE THREAD

**Issue by jf** [jcupitt/ruby-vips8#1]
Sorry, but I am not getting this gem. I've got libvips installed (and even ensured that the vips library is accessible; see https://github.com/jcupitt/libvips/issues/389). Now I install this gem as per the instructions here (simply `gem install ruby-vips8`), and try to get things going, only to discover that I am apparently missing some typelib?

Sorry, but what is a typelib, and how and where do I get `Vips-8.0.typelib`? I've tried searching in my copy of the libvips source, after the compile, but a `find |grep typelib` in that directory reveals nothing.

One of the first things I tried was the example given in the README:

```
require 'vips8'

im = Vips::Image.new_from_file '/tmp/somefilename.jpg'
...
```

but this immediately fails:

```
irb(main):001:0> require 'vips8'
=> true
irb(main):002:0> Vips::Image.new_from_filename '/tmp/somefilename.jpg'
Unable to load Vips
  Check that the vips library has been installed and is
  on your library path.
  Check that the typelib `Vips-8.0.typelib` has been
  installed, and that it is on your GI_TYPELIB_PATH.
GLib::Error: Typelib file for namespace 'Vips' (any version) not found
        from /home/jf/.rbenv/versions/2.2.3/gemsets/x/gems/gobject-introspection-3.0.7/lib/gobject-introspection/loader.rb:37:in `require'
        from /home/jf/.rbenv/versions/2.2.3/gemsets/x/gems/gobject-introspection-3.0.7/lib/gobject-introspection/loader.rb:37:in `load'
        from /home/jf/.rbenv/versions/2.2.3/gemsets/x/gems/ruby-vips8-0.1.0/lib/vips8.rb:79:in `init'
        from /home/jf/.rbenv/versions/2.2.3/gemsets/x/gems/ruby-vips8-0.1.0/lib/vips8.rb:47:in `const_missing'
        from (irb):2
        from /home/jf/.rbenv/versions/2.2.3/bin/irb:11:in `<main>'
irb(main):003:0> 
```

Doing a search, I come across http://www.vips.ecs.soton.ac.uk/supported/current/doc/html/libvips/using-from-python.html, which mentions a "libvips typelib" as well; but also offers no concrete help as to where to find this. A search from the main page (which redirects to a wiki system), with http://www.vips.ecs.soton.ac.uk/index.php?title=Special%3ASearch&search=typelib&go=Go also reveals nothing. I try the blog as well (http://libvips.blogspot.com/search?q=typelib), but find nothing. How do I resolve this, and get the gem working?

**Comment by jcupitt** [comment]
Hi @jf, sorry it's not working.

The main docs page for the Vips class has some background:

http://www.rubydoc.info/gems/ruby-vips8/0.1.0/Vips

Scroll down to "How it works". I would check:

1. After you run `./configure` in libvips, check that it's found gobject-introspection. The summary at the end should display:

```
* build options
native win32:				no
native OS X:				no
open files in binary mode: 		no
enable debug:				yes
build deprecated components:		yes
build docs with gtkdoc:			no
gobject introspection: 			yes
build vips7 Python binding:		yes
install vips8 Python overrides:		yes
  (requires pygobject-3.12.0 or later)
```

ie. you need a `yes` next to `gobject-introspection`, if you don't have that, install the goi packages. 

2. Next, after `make install`, check that a typelib was built. It will be installed to `prefix/lib/girepository-1.0/`, so if you install to `/usr/local`, check `/usr/local/lib/girepository-1.0/`

3. Set your `GI_TYPELIB_PATH` to include this directory. It may include it already, depending on your system.

And that ought to be it, hopefully. Good luck!

**Comment by jcupitt** [comment]
This should become much easier when libvips 8.2 gets packaged, it'll just be two commands. ruby-vips8 is still very young and support hasn't spread though the system yet.

**Comment by jf** [comment]
Ok. Thanks for the very clear and specific instructions. Apologies about me missing `gobject-introspection`: I ignored it before building as it seemed from the `libvips` README that it was an optional dependency, and then only used in connection with Python. And since I wasn't going to be using Python....

So now after recompiling and reinstalling I have `Vips-8.0.typelib` compiled and installed to `/usr/local/lib/girepository-1.0/` now, which is great. I still got the same error though after attempting to use the gem, but this time since I know where it is installed, I surmised that an `export GI_TYPELIB_PATH=/usr/local/lib/girepository-1.0` would do, and indeed it seems to have done so. I can now get past the `new_from_file`, and I assume I don't need to worry about getting this error again... Thanks!
